File: JavaScript/03-Lexical-Scope-Closures/counter.js
function counter() {
  let count = 0;

  return function () {
    return ++count;
  };
}

const c1 = counter();
console.log(c1()); // 1
console.log(c1()); // 2
console.log(c1()); // 3
console.log(c1()); // 4

File: DSA/Intermediate-Refresher/TrickyQuestions/Main-Method/Counter.java
public class Counter {

    static int count = 1;

    public static void main(String[] args) {

        System.out.println("Execution: " + count);

        if (count < 5) {
            count++;
            main(new String[] {});
        }

        System.out.println("Returning from call " + count);
        count--;
    }
}
